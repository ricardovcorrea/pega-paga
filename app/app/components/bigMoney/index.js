import React from 'react';
import {View, Text} from 'react-native';

import {formatAsCurrency} from '../../general/helpers';
import styles from './styles';

const BigMoney = ({title, value}) => {
  const formatedValue = formatAsCurrency(value / 100);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.value}>{formatedValue}</Text>
    </View>
  );
};

export default BigMoney;
