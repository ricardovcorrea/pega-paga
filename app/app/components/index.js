export {default as BaseScreen} from './baseScreen/baseScreen.js';
export {default as Button} from './button/button.js';
export {default as TextInput} from './textInput/index';
export {default as TabBar} from './tabBar/index';
export {default as UserProfileCard} from './userProfileCard/index';
export {default as BigMoney} from './bigMoney/index';
export {default as NavBarLogo} from './navBarLogo/index';
