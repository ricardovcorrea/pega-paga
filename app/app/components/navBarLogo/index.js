import React from 'react';
import { StyleSheet, Text } from 'react-native';

const styles = StyleSheet.create({
    logo: {
        fontSize: 30,
        fontWeight: 'bold',
        color: 'white',
        padding: 10
    }
});

const NavBarLogo = props => {
    return (
        <Text style={styles.logo}>Pega Paga</Text>
    )
}

export default NavBarLogo;