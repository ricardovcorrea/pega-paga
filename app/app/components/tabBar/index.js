import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity
} from 'react-native';

import MaterialIcons from "react-native-vector-icons/MaterialIcons";

import { isIPhoneX } from '../../general/helpers';

import theme from '../../general/theme';

const styles = StyleSheet.create({
    container: { flexDirection: "row", height: 52, elevation: 2, borderTopWidth: 1, borderTopColor: '#e5e5e5', backgroundColor: '#fff', marginBottom: isIPhoneX() ? 15 : 0 },
    tabButton: { flex: 1, justifyContent: "center", alignItems: "center" },
    title: { fontSize: 11, marginTop: 4 },
    badge: { width: 13, height: 13, borderRadius: 6, backgroundColor: '#e5b900', position: 'absolute', right: -5, top: -4 }
});

const TabBar = (props) => {

    const {
        renderIcon,
        getLabelText,
        activeTintColor,
        inactiveTintColor,
        onTabPress,
        onTabLongPress,
        getAccessibilityLabel,
        navigation
    } = props;

    const { routes, index: activeRouteIndex } = navigation.state;

    const tabLabelAndIcon = {
        Main: {
            icon: 'home',
            label: 'Home'
        },
        Pay: {
            icon: 'filter-center-focus',
            label: 'Pay'
        },
        Receive: {
            icon: 'attach-money',
            label: 'Receive'
        },
        Menu: {
            icon: 'menu',
            label: 'Menu'
        }
    }

    return (
        <View style={styles.container}>
            {routes.map((route, routeIndex) => {
                const { routeName } = route;
                const isRouteActive = routeIndex === activeRouteIndex;
                const tintColor = isRouteActive ? activeTintColor : inactiveTintColor;

                return (
                    <TouchableOpacity
                        key={routeIndex}
                        style={styles.tabButton}
                        onPress={() => {
                            onTabPress({ route });
                        }}
                        accessibilityLabel={getAccessibilityLabel({ route })}>

                        <>
                            <MaterialIcons name={tabLabelAndIcon[routeName].icon} size={30} color={isRouteActive ? theme.primary : inactiveTintColor} />
                            <Text style={[styles.title, isRouteActive ? { color: tintColor } : { color: inactiveTintColor }]}>{tabLabelAndIcon[routeName].label}</Text>
                        </>

                    </TouchableOpacity>
                );
            })}
        </View>
    );
}


export default TabBar;