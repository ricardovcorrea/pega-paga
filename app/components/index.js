export {default as BaseScreen} from './baseScreen/baseScreen.js';
export {default as Button} from './button/button.js';
export {default as TextInput} from './textInput/index';
export {default as TabBar} from './tabBar/tabBar';
export {default as UserProfileCard} from './userProfileCard/index';
export {default as BigMoney} from './bigMoney/bigMoney';
export {default as NavBarLogo} from './navBarLogo/index';
export {default as TransactionsList} from './transactionList/index';
