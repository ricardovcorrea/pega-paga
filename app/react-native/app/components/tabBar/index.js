import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

import theme from '~/general/theme';
import styles from './styles';

const TabBar = props => {
  const {
    activeTintColor,
    inactiveTintColor,
    onTabPress,
    getAccessibilityLabel,
    navigation,
  } = props;

  const {routes, index: activeRouteIndex} = navigation.state;

  const tabLabelAndIcon = {
    Main: {
      icon: 'home',
      label: 'Home',
    },
    Pay: {
      icon: 'filter-center-focus',
      label: 'Pay',
    },
    Receive: {
      icon: 'attach-money',
      label: 'Receive',
    },
    Menu: {
      icon: 'menu',
      label: 'Menu',
    },
  };

  return (
    <View style={styles.container}>
      {routes.map((route, routeIndex) => {
        const {routeName} = route;
        const isRouteActive = routeIndex === activeRouteIndex;
        const tintColor = isRouteActive ? activeTintColor : inactiveTintColor;

        return (
          <TouchableOpacity
            key={routeIndex}
            style={styles.tabButton}
            onPress={() => {
              onTabPress({route});
            }}
            accessibilityLabel={getAccessibilityLabel({route})}>
            <>
              <MaterialIcons
                name={tabLabelAndIcon[routeName].icon}
                size={30}
                color={isRouteActive ? theme.primary : inactiveTintColor}
              />
              <Text
                style={[
                  styles.title,
                  isRouteActive
                    ? {color: tintColor}
                    : {color: inactiveTintColor},
                ]}>
                {tabLabelAndIcon[routeName].label}
              </Text>
            </>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default TabBar;
