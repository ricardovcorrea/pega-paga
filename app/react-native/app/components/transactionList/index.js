/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable prettier/prettier */
import moment from 'moment';
import React, {useState, useEffect} from 'react';
import {View, Text, FlatList} from 'react-native';

import {formatAsCurrency} from '~/general/helpers';

import styles from './styles';

const TransactionsList = props => {
  const {userId} = props;
  const [formatedTransactions, setFormatedTransactions] = useState([]);

  useEffect(() => {
    formatTransactionsData();
  }, [props.transactions]);

  const formatTransactionsData = () => {
    if (!props.transactions || !props.transactions.length) {
      return;
    }

    let formatedData = [];

    let todayTransactions = [];
    const todayStart = moment().startOf('day');

    let yesterdayTransactions = [];
    const yesterdayStart = moment()
      .add(-1, 'd')
      .startOf('day');

    let sevenDaysTransactions = [];
    const sevenDaysStart = moment()
      .add(-7, 'd')
      .startOf('day');

    let allOtherTransacions = [];
    const allOtherTransacionsStart = moment()
      .add(-10, 'Y')
      .startOf('day');

    todayTransactions = props.transactions.filter(
      transaction => moment(transaction.datetime) >= todayStart,
    );

    yesterdayTransactions = props.transactions.filter(
      transaction =>
        moment(transaction.datetime) >= yesterdayStart &&
        moment(transaction.datetime) < todayStart,
    );

    sevenDaysTransactions = props.transactions.filter(
      transaction =>
        moment(transaction.datetime) >= sevenDaysStart &&
        moment(transaction.datetime) < yesterdayStart,
    );

    allOtherTransacions = props.transactions.filter(
      transaction =>
        moment(transaction.datetime) >= allOtherTransacionsStart &&
        moment(transaction.datetime) < sevenDaysStart,
    );

    if (todayTransactions.length) {
      formatedData.push({
        type: 'divider',
        props: {
          title: 'Today',
        },
      });

      formatedData = [
        ...formatedData,
        ...todayTransactions.map(transaction => ({
          type: 'transaction',
          transaction,
        })),
      ];
    }

    if (yesterdayTransactions.length) {
      formatedData.push({
        type: 'divider',
        props: {
          title: 'Yesterday',
        },
      });

      formatedData = [
        ...formatedData,
        ...yesterdayTransactions.map(transaction => ({
          type: 'transaction',
          transaction,
        })),
      ];
    }

    if (sevenDaysTransactions.length) {
      formatedData.push({
        type: 'divider',
        props: {
          title: 'Week',
        },
      });

      formatedData = [
        ...formatedData,
        ...sevenDaysTransactions.map(transaction => ({
          type: 'transaction',
          transaction,
        })),
      ];
    }

    if (allOtherTransacions.length) {
      formatedData.push({
        type: 'divider',
        props: {
          title: 'All other',
        },
      });

      formatedData = [
        ...formatedData,
        ...allOtherTransacions.map(transaction => ({
          type: 'transaction',
          transaction,
        })),
      ];
    }

    setFormatedTransactions(formatedData);
  };

  const getStickyToHeaderDividerIndexList = () => {
    const indexsList = formatedTransactions
      .map((data, index) => (data.type === 'divider' ? index : -1))
      .filter(index => index > -1);
    return indexsList;
  };

  const renderTransaction = itemToRender => {
    const {transaction} = itemToRender;
    return (
      <View style={styles.transactionItemContainer}>
        {transaction.from.id === userId && (
          <Text>{`${transaction.to.firstName} ${transaction.to.surname}`}</Text>
        )}
        {transaction.to.id === userId && (
          <Text>{`${transaction.from.firstName} ${transaction.from.surname}`}</Text>
        )}
        <Text>{`${moment(transaction.datetime).format('DD/MM')} at ${moment(
          transaction.datetime,
        ).format('HH:MM')} `}</Text>

        <View style={styles.transactionItemAmountContainer}>
          {transaction.from.id === userId && <Text>-</Text>}
          {transaction.to.id === userId && <Text>+</Text>}
          <Text>{formatAsCurrency(transaction.amount / 100)}</Text>
        </View>
      </View>
    );
  };

  const renderDivider = itemToRender => {
    const dividerProps = itemToRender.props;

    return (
      <View style={styles.transactionItemDivider}>
        <Text style={styles.transactionItemDividerText}>
          {dividerProps.title}
        </Text>
        <Text style={styles.transactionItemDividerText}>
          {dividerProps.balance && formatAsCurrency(dividerProps.balance / 100)}
        </Text>
      </View>
    );
  };

  const renderTransactionListItem = item => {
    const itemToRender = item.item;
    if (itemToRender.type === 'transaction') {
      return renderTransaction(itemToRender);
    }
    if (itemToRender.type === 'divider') {
      return renderDivider(itemToRender);
    }
  };

  return (
    <FlatList
      showsVerticalScrollIndicator={false}
      contentContainerStyle={styles.transactionListContainer}
      style={styles.transactionList}
      data={formatedTransactions}
      keyExtractor={(transaction, index) => index.toString()}
      stickyHeaderIndices={getStickyToHeaderDividerIndexList()}
      renderItem={renderTransactionListItem}
    />
  );
};

export default TransactionsList;
