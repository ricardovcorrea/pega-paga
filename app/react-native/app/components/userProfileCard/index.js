import React from 'react';
import {View, Text, Image} from 'react-native';

import styles from './styles';

const UserProfileCard = props => {
  const {id, firstName, surname, email, photo} = props.user;

  return (
    <View style={[styles.container, props.style]}>
      <Image
        source={{
          uri:
            photo ||
            'https://st2.depositphotos.com/4967775/11323/v/950/depositphotos_113235752-stock-illustration-avatar-girls-icon-vector-woman.jpg',
        }}
        style={styles.image}
      />
      <View style={styles.textContainer}>
        <Text style={styles.textId}>ID: {id}</Text>
        <Text style={styles.textFirstName}>
          {`${firstName}, `}
          <Text style={styles.textEmail}>{surname}</Text>
        </Text>
        <Text style={styles.textEmail}>{email}</Text>
      </View>
    </View>
  );
};

export default UserProfileCard;
