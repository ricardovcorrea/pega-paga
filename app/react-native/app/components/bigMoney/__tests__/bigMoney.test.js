import React from 'react';
import renderer from 'react-test-renderer';

import {BigMoney} from '~/components';

describe('Big Money Component Tests', () => {
  it('should render itself with title and formated value', () => {
    const component = renderer.create(
      <BigMoney title={'Title'} value={3000} />,
    );
    expect(component).toMatchSnapshot();
  });
});
