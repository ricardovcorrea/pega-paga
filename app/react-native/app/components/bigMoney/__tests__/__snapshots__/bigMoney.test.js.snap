// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big Money Component Tests should render itself with title and formated value 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "600",
        "marginBottom": 15,
      }
    }
  >
    Title
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 90,
        "fontWeight": "900",
      }
    }
  >
    R$Â 30.00
  </Text>
</View>
`;
